File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>
#include <queue>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    int n = arr.size();
    queue<int> q;
    
    if(n!=0) {
        int x = arr[0];
        q.push(x);
    
        for(int i=1; i<n; i++){
            if(x!=arr[i]){
                q.push(arr[i]);
                x=arr[i];
            }
        }
    }
    
    int s = q.size();
    if(s!=0){
        for(int i=0; i<s; i++){
            answer.push_back(q.front());
            q.pop();
        }
    }
    
    return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        Queue<Integer> queue = new LinkedList<>(); //큐는 LinkedList 형태임
        
        if(arr.length != 0){
            int prev = arr[0];
            queue.offer(prev); //큐에 값 넣기
            
            for(int i=1; i<arr.length; i++){
                if(arr[i] != prev){
                    queue.offer(arr[i]);
                    prev = arr[i];
                }
            }
        }
        
        int i = 0;
        int[] answer = new int[queue.size()];
        while(!queue.isEmpty()){
            answer[i++] = queue.poll(); //C++과 다른 점, i++과 ++i의 차이
        }

        return answer;
    }
}